### PR TITLE
manifest: openthread upmerge up to commit `2ce3d3b`

### PR DESCRIPTION
**Pull request by edmont**
Perform OpenThread upmerge and enable Thread 1.3 by default.
Remove Border Router support from Master feature set.
Include needed Matter changes.
Include needed HomeKit changes.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

### CODE REVIEW

**Comment by kkasperczyk-no** [comment]
@edmont please update the sdk-connectedhomeip SHA in the manifest to the `5959a2ed66563f5f34cb6229cba4698e460f57d4` and also commit SHA in documentation links: https://github.com/nrfconnect/sdk-nrf/blob/v2.0-branch/doc/nrf/links.txt#L93, https://github.com/nrfconnect/sdk-nrf/blob/v2.0-branch/doc/nrf/links.txt#L94.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -69,6 +69,8 @@ The following list summarizes the most important changes inherited from the upst
 Thread
 ------
 
+* Enabled Thread 1.3 features by default.
+* Removed support for Thread Border Router as part of the Master library feature set.
 * Fixed a bug in which a Minimal Thread Device was not able to handle Address Error Notification messages.
 * Updated the values in the memory requirement tables in :ref:`thread_ot_memory` after the update to the :ref:`nrfxlib:ot_libs` in nrfxlib.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:75]: These belong to the previous release, don't they?
```suggestion

```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (edmont) [reply doc/nrf/releases/release-notes-changelog.rst:75]: You are right.

Comment (edmont) [reply doc/nrf/releases/release-notes-changelog.rst:75]: Agreed to remove on a separate PR.

**Review by rlubos — APPROVED** [review]

**Review by kkasperczyk-no — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by canisLupus1313 — APPROVED** [review]

**Pull request merged**